authorization/invitation/service/invitation_service_blackbox_test.go

```diff
@@ -34,619 +38,693 @@ func (s *invitationServiceBlackBoxTest) SetupTest() {
+			},
 		}
-	}
-	require.True(s.T(), found, "Second invitee not found in invitations")
-}
 
-func (s *invitationServiceBlackBoxTest) TestIssueInvitationByIdentityIDForRole() {
-	team := s.Graph.CreateTeam()
-	teamAdmin := s.Graph.CreateUser()
-	user := s.Graph.CreateUser()
-	r := s.Graph.CreateRole(s.Graph.LoadResourceType(authorization.IdentityResourceTypeTeam))
-	r.AddScope(authorization.ManageTeamMembersScope)
+		err = s.Application.InvitationService().Issue(s.Ctx, otherIdentity.ID, orgId.String(), invitations)
+		require.Error(s.T(), err)
+	})
 
-	team.AssignRole(teamAdmin.Identity(), r.Role())
+	s.T().Run("should fail to issue invitation for unknown user", func(t *testing.T) {
+		// Create a test user - this will be the organization owner
+		identity, err := test.CreateTestIdentityAndUserWithDefaultProviderType(s.DB, "invitationServiceBlackBoxTest-TestIssuingUser")
+		require.Nil(s.T(), err, "Could not create identity")
 
-	id := user.IdentityID()
+		// Create an organization
+		orgId, err := s.orgService.CreateOrganization(s.Ctx, identity.ID, "Test Organization "+uuid.NewV4().String())
+		require.Nil(s.T(), err, "Could not create organization")
 
-	invitations := []invitation.Invitation{
-		{
-			IdentityID: &id,
-			Roles:      []string{r.Role().Name},
-			Member:     false,
-		},
-	}
+		invalidIdentityID := uuid.NewV4()
 
-	err := s.Application.InvitationService().Issue(s.Ctx, teamAdmin.IdentityID(), team.TeamID().String(), invitations)
-	require.NoError(s.T(), err, "Error creating invitations")
+		invitations := []invitation.Invitation{
+			{
+				IdentityID: &invalidIdentityID,
+				Member:     true,
+			},
+		}
 
-	invs, err := s.invitationRepo.ListForIdentity(s.Ctx, team.TeamID())
-	require.NoError(s.T(), err, "Error listing invitations")
+		// This should fail because we specified an unknown identity ID
+		err = s.Application.InvitationService().Issue(s.Ctx, identity.ID, orgId.String(), invitations)
+		require.Error(s.T(), err)
+	})
+
+	s.T().Run("should fail to issue invitation for non user", func(t *testing.T) {
+		// Create a test user - this will be the organization owner
+		identity, err := test.CreateTestIdentityAndUserWithDefaultProviderType(s.DB, "invitationServiceBlackBoxTest-TestIssuingUser")
+		require.Nil(s.T(), err, "Could not create identity")
+
+		// Create an organization, we're going to do something crazy and invite the organization to join itself
+		orgId, err := s.orgService.CreateOrganization(s.Ctx, identity.ID, "Test Organization "+uuid.NewV4().String())
+		require.Nil(s.T(), err, "Could not create organization")
+
+		invitations := []invitation.Invitation{
+			{
+				IdentityID: orgId,
+				Member:     true,
+			},
+		}
 
-	require.Len(s.T(), invs, 1)
-	require.False(s.T(), invs[0].Member)
+		// This should fail because we specified a non-user identity in the invitation
+		err = s.Application.InvitationService().Issue(s.Ctx, identity.ID, orgId.String(), invitations)
+		require.Error(s.T(), err)
```

Comment:
same here again: could you also verify that the NotificationService was not called at all, in this test?